graphql-dgs-subscriptions-sse/src/main/kotlin/com/netflix/graphql/dgs/subscriptions/sse/DgsSSESubscriptionHandler.kt

```diff
@@ -120,10 +126,26 @@ open class DgsSSESubscriptionHandler(open val dgsQueryExecutor: DgsQueryExecutor
                 Flux.just(
                     ServerSentEvent.builder(mapper.writeValueAsString(payload))
                         .id(UUID.randomUUID().toString())
-                        .event("error")
```

Comment:
I think this is the main change that may lead to breaking existing clients. 

The original SSE implementation in the framework is based on subscriptions-transport-sse : https://github.com/CodeCommission/subscriptions-transport-sse . This was prior to the introduction of graphql-sse protocol described here: https://github.com/enisdenjo/graphql-sse/blob/master/PROTOCOL.md.

This seems to be the source of discrepancy. Your changes look good to me, but I think it would be better to introduce this as a new module to opt into call dgs-subscriptions-graphql-sse. This will prevent breakage of existing clients and also allow folks who want to use this newer standard SSE protocol to opt in.

With that strategy, we should be able to avoid making this a breaking release.

Reply:
I have made a new package called graphql-dgs-subscriptions-graphql-sse I have also created a brand new graphql client so that users can specifically choose which client they need depending on the server implementation